include/boost/math/special_functions/detail/fp_traits.hpp

```diff
@@ -25,8 +25,36 @@ With these techniques, the code could be simplified.
 #include <limits>
 #include <type_traits>
 #include <boost/math/tools/assert.hpp>
+
+// Determine endinaness
+#ifndef BOOST_MATH_STANDALONE
+
 #include <boost/predef/other/endian.h>
-#include <boost/math/tools/assert.hpp>
+#define BOOST_MATH_ENDIAN_BIG_BYTE BOOST_ENDIAN_BIG_BYTE
+#define BOOST_MATH_ENDIAN_LITTLE_BYTE BOOST_ENDIAN_LITTLE_BYTE
+
+#else // Using standalone mode
+#if (__cplusplus > 202000L || _MSVC_LANG > 202000L) && __has_include(<bit>)
+
+#include <bit>
+#define BOOST_MATH_ENDIAN_BIG_BYTE (std::endian::native == std::endian::big)
+#define BOOST_MATH_ENDIAN_LITTLE_BYTE (std::endian::native == std::endian::little)
+
+#else // Does not have compliant C++20
+
+#ifdef _WIN32
+
+#define BOOST_MATH_ENDIAN_BIG_BYTE 1
+#define BOOST_MATH_ENDIAN_LITTLE_BYTE 0
+
+#else
+
+#define BOOST_MATH_ENDIAN_BIG_BYTE (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
```

Comment:
I think this is nearly but not quite correct: these macros are GCC specific I assume?  We should probably have another #elif branch here for the case where `__BYTE_ORDER__` is not defined and a reasonably helpful #error message explaining that either the header needs to be fixed, or they should not use standalone mode.

Reply:
Those macros work with Clang as well. Adding an error message if they are not defined makes sense.